src/__tests__/use-replicant.tsx

```diff
@@ -18,44 +30,119 @@ class Replicant extends EventEmitter {
-	const [currentRun] = useReplicant('currentRun', {runner: {name: 'foo'}});
+interface RunnerNameProps {
+	prefix?: string;
+}
+
+const RunnerName: React.FC<RunnerNameProps> = (props): JSX.Element => {
+	const {prefix} = props;
+	const repName = `${prefix || 'default'}:currentRun`;
+	const [currentRun] = useReplicant(repName, {runner: {name: 'foo'}});
 	return <div>{currentRun.runner.name}</div>;
 };
 
+// Example of a replicant with a mutating value.
+const Counter: React.FC = (): JSX.Element => {
+	const [counter, setCounter] = useReplicant('counter', 0);
+	return <button onClick={() => setCounter(counter + 1)}>{counter}</button>;
+};
+
 let renderResult: RenderResult;
 beforeEach(() => {
-	renderResult = render(<RunnerName />);
+	allReplicants.clear();
+	replicantHandler.mockReset();
+	replicantRemoveListener.mockReset();
 });
 
 test('Initializes replicant correctly', () => {
-	expect(replicantConstructor).toBeCalledWith('currentRun', {
+	renderResult = render(<RunnerName />);
+	expect(replicantConstructor).toBeCalledWith('default:currentRun', {
 		defaultValue: {
 			runner: {name: 'foo'},
 		},
 	});
 });
 
 test('Change handler is set correctly', () => {
+	renderResult = render(<RunnerName />);
 	expect(replicantHandler).toBeCalledTimes(1);
 });
 
-test.skip('Handles replicant changes', () => {
-	replicant.emit('change', {runner: {name: 'bar'}});
+test('Change not triggered on rerender', () => {
+	renderResult = render(<RunnerName />);
+	const timesCalled = replicantHandler.mock.calls.length;
+	renderResult.rerender(<RunnerName />);
+	expect(replicantHandler).toBeCalledTimes(timesCalled);
+});
+
+test('Handles replicant name changes', () => {
+	renderResult = render(<RunnerName />);
+	const timesCalled = replicantHandler.mock.calls.length;
+	renderResult.rerender(<RunnerName prefix='test2' />);
+	expect(replicantConstructor).toBeCalledWith('test2:currentRun', {
+		defaultValue: {
+			runner: {name: 'foo'},
+		},
+	});
+	expect(replicantHandler).toBeCalledTimes(timesCalled + 1);
+});
+
+test('Handles replicant changes', () => {
+	renderResult = render(<RunnerName />);
+	expect(allReplicants.size).toEqual(1);
+	const replicant = allReplicants.values().next().value;
+	act(() => {
+		replicant.emit('change', {runner: {name: 'bar'}});
+	});
 	renderResult.rerender(<RunnerName />);
 	expect(renderResult.container.textContent).toBe('bar');
 });
 
+test('Can change replicant value using hook', () => {
+	renderResult = render(<Counter />);
+	expect(allReplicants.has('counter')).toBe(true);
+	expect(renderResult.container.firstChild).toBeTruthy();
+	const replicant = allReplicants.get('counter');
+
+	// We know its not undefined cause we asserted it, but we have a picky linter.
+	if (typeof replicant === 'undefined') return;
+	if (renderResult.container.firstChild === null) return;
+
+	const initialValue = replicant.value as number;
+	expect(replicant.value).toBe(0);
+
+	fireEvent.click(renderResult.container.firstChild as Element);
```

Comment:
Pre-emptively gonna state, despite fireEvent.click(container.firstChild) is the recommended way of doing it in the tutorial, the type system really didn't like it. Therefore I casted it. If you know of an alternative way of doing this then it'd certainly help a lot.

One alternative might be using the react testing library specifically for hooks. Not sure about doing that though.